x-pack/plugins/cases/public/components/app/routes.tsx

```diff
@@ -45,47 +50,51 @@ const CasesRoutesComponent: React.FC<CasesRoutesProps> = ({
   );
 
   return (
-    <Switch>
-      <Route strict exact path={basePath}>
-        <AllCases />
-      </Route>
+    <QueryClientProvider client={casesQueryClient}>
+      <Switch>
+        <Route strict exact path={basePath}>
+          <AllCases />
+        </Route>
 
-      <Route path={getCreateCasePath(basePath)}>
-        {userCanCrud ? (
-          <CreateCase
-            onSuccess={onCreateCaseSuccess}
-            onCancel={navigateToAllCases}
-            timelineIntegration={timelineIntegration}
-          />
-        ) : (
-          <NoPrivilegesPage pageName={i18n.CREATE_CASE_PAGE_NAME} />
-        )}
-      </Route>
+        <Route path={getCreateCasePath(basePath)}>
+          {userCanCrud ? (
+            <CreateCase
+              onSuccess={onCreateCaseSuccess}
+              onCancel={navigateToAllCases}
+              timelineIntegration={timelineIntegration}
+            />
+          ) : (
+            <NoPrivilegesPage pageName={i18n.CREATE_CASE_PAGE_NAME} />
+          )}
+        </Route>
 
-      <Route path={getCasesConfigurePath(basePath)}>
-        {userCanCrud ? (
-          <ConfigureCases />
-        ) : (
-          <NoPrivilegesPage pageName={i18n.CONFIGURE_CASES_PAGE_NAME} />
-        )}
-      </Route>
+        <Route path={getCasesConfigurePath(basePath)}>
+          {userCanCrud ? (
+            <ConfigureCases />
+          ) : (
+            <NoPrivilegesPage pageName={i18n.CONFIGURE_CASES_PAGE_NAME} />
+          )}
+        </Route>
 
-      <Route exact path={[getCaseViewWithCommentPath(basePath), getCaseViewPath(basePath)]}>
-        <CaseView
-          onComponentInitialized={onComponentInitialized}
-          actionsNavigation={actionsNavigation}
-          ruleDetailsNavigation={ruleDetailsNavigation}
-          showAlertDetails={showAlertDetails}
-          useFetchAlertData={useFetchAlertData}
-          refreshRef={refreshRef}
-          timelineIntegration={timelineIntegration}
-        />
-      </Route>
+        <Route exact path={[getCaseViewWithCommentPath(basePath), getCaseViewPath(basePath)]}>
+          <Suspense fallback={<EuiLoadingSpinner />}>
+            <CaseViewLazy
```

Comment:
moving the caseview to a lazy component because it will be too big with react-query included.